src/utils/fetch-products.ts

```diff
@@ -0,0 +1,9 @@
+import { products } from '@/constants/products.constant.ts';
+import type { Product } from '@/models/product.model.ts';
+
+export const fetchProducts = (): Promise<Product[]> => {
```

Comment:
I recommend you to create your custom hook for fetching data. You may find out about it in the articles below.

https://codesandbox.io/p/sandbox/useapi-react-hook-pwgyl?file=%2Fsrc%2Fhooks%2FuseApi.js
https://dou.ua/forums/topic/47858/